evm/hyperliquid.js

```diff
@@ -0,0 +1,225 @@
+'use strict';
+
+const { Wallet, ethers, getDefaultProvider, Contract, AddressZero, BigNumber } = require('ethers');
+const { exec } = require('child_process');
+const { promisify } = require('util');
+const { Command, Option } = require('commander');
+const { printInfo, validateParameters, getContractJSON, getGasOptions, mainProcessor, isHyperliquidChain } = require('./utils');
+const { addEvmOptions, addOptionsToCommands } = require('./cli-utils');
+const { httpPost } = require('../common/utils');
+const { handleTx } = require('./its');
+const execAsync = promisify(exec);
+const msgpack = require('msgpack-lite');
+const { keccak256 } = require('ethers/lib/utils');
+
+function addressToBytes(address) {
+    return Buffer.from(address.replace('0x', ''), 'hex');
+}
+
+function actionHash(action, activePool, nonce) {
+    const actionData = msgpack.encode(action);
+    const nonceBuffer = Buffer.alloc(8);
+    nonceBuffer.writeBigUInt64BE(BigInt(nonce));
+
+    const vaultBuffer =
+        activePool === null || activePool === undefined
+            ? Buffer.from([0x00])
+            : Buffer.concat([Buffer.from([0x01]), addressToBytes(activePool)]);
+
+    const data = Buffer.concat([actionData, nonceBuffer, vaultBuffer]);
+    return keccak256(data);
+}
+
+function constructPhantomAgent(hash, source) {
+    return {
+        source: source,
+        connectionId: hash,
+    };
+}
+
+async function signL1Action(wallet, action, activePool, nonce, chain) {
+    const hash = actionHash(action, activePool, nonce);
+    const phantomAgent = constructPhantomAgent(hash, chain.hypercore?.source);
+    const domain = chain.hypercore?.domain;
```

Comment:
can you add all these hypercore configs to the corresponding hyperliquid testnet/mainnet PRs?